concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -253,6 +256,39 @@ type DirectMessageCallback = extern "C" fn(
 type RegenesisCallback = unsafe extern "C" fn(*const Regenesis, *const u8);
 type RegenesisFreeCallback = unsafe extern "C" fn(*const Regenesis);
 
+/// A type of callback that extends the given vector with the provided data.
+type CopyToVecCallback = extern "C" fn(*mut Vec<u8>, *const u8, i64);
+
+/// Context necessary for Haskell code/Consensus to send notifications on
+/// important events to Rust code (i.e., RPC server, or network layer).
+pub struct NotificationContext {
+    /// Notification channel for newly added blocks. This is an unbounded
+    /// channel since it makes the implementation simpler. This should not be a
+    /// problem since the consumer of this channel is a dedicated task and
+    /// blocks are not added to the tree that quickly. So there should not be
+    /// much contention for this.
```

Comment:
What about when'catching up' ?

Reply:
It takes tens of milliseconds to process a block. That's an eternity.

In any case, I have tested it during catchup :P